application/models/User.php

```diff
@@ -970,17 +965,40 @@ public function getGroupMemberListButtons()
     }
 
     /**
-     * Returns true if logged in user with id $loginId can edit this user
+     * Return true if user with id $managerId can edit this user
+     * @param int|null $managerId default to current user
      *
-     * @param int $loginId
      * @return bool
      */
-    public function canEdit($loginId)
+    public function canEdit($managerId = null)
     {
-        return
-            Permission::model()->hasGlobalPermission('superadmin', 'read')
-            || $this->uid == $loginId
-            || (Permission::model()->hasGlobalPermission('users', 'update') && $this->parent_id == $loginId);
+        if (is_null($managerId)) {
```

Comment:
Is this code to be here or on the UserManager? @olleharstedt What do you say?

Reply:
> Is this code to be here or on the UserManager? @olleharstedt What do you say?

This code in the 1st commit was in UserManager but it broke API …

The only way to don't broke API is to do here or to create a new function …

Reply:
If you user UserManager : you can not have `canEdit($loginId)` because i can not have a App()->user with only id.

Then    `canEdit($loginId)`  CAN NOT use UserManager https://github.com/LimeSurvey/LimeSurvey/blob/f5e3a0f252bae70ec89ba76e1ac938e12df729ac/application/models/services/UserManager.php#L16 

And the original issue are done because we have 3 functions for same usage …

Reply:
> If you user UserManager : you can not have `canEdit($loginId)` because i can not have a App()->user with only id.

Not sure I follow, but I was thinking something like this:

On `getManagementButtons()` you could have:

```
$userManager = new UserManager($this, App()->user->getId())
$canEdit = $userManager->canEdit();
```

Still, not sure these back and forths wort too much. Let's just wait for @olleharstedt to let us know where he thinks the best palce for that logic is :)

Reply:
> > If you user UserManager : you can not have `canEdit($loginId)` because i can not have a App()->user with only id.
> 
> Not sure I follow, but I was thinking something like this:
> 
> On `getManagementButtons()` you could have:
> 
> ```
> $userManager = new UserManager(App()->user, $this)
> $canEdit = $userManager->canEdit();
> ```
> 
> Still, not sure these back and forths wort too much. Let's just wait for @olleharstedt to let us know where he thinks the best palce for that logic is :)

Yes : this make https://github.com/LimeSurvey/LimeSurvey/blob/73f26617a5d149a74d611697721fbfda8813c332/application/models/User.php#L978 unusable like this. Then it broke API

I start with this solution : look at https://github.com/LimeSurvey/LimeSurvey/pull/3495/commits/dee58d835c3975d1ebf8fe7141ea603556d27700 and i update after your advice about API version (updating a public function signature)

Reply:
> Your sample is exactly the new function can edit : https://github.com/LimeSurvey/LimeSurvey/commit/dee58d835c3975d1ebf8fe7141ea603556d27700#diff-0a0e2b974470e717a216950e1374f76ba04f409a55f562a9b59f91cb79451ccdR974

Great. Not sure if you are expecting any action from me here or not, sorry